src/Compilation.zig

```diff
@@ -1081,6 +1084,9 @@ pub fn addSourceFromReader(comp: *Compilation, reader: anytype, path: []const u8
     const splice_locs = try splice_list.toOwnedSlice();
     errdefer comp.gpa.free(splice_locs);
 
+    // Important: do not perform any possibly-failing operations in this function after this realloc.
+    // Otherwise errdefers in callers will possibly free the realloced slice using the original len
+    // instead of the new
```

Comment:
Check this out, you can put this on the next line:
```zig
errdefer @compileError("errdefers in callers would possibly free the realloced slice using the original len");
```

Reply:
oh wow, that's awesome!